decompress.c

```diff
@@ -65,6 +65,68 @@ static sqfs_err sqfs_decompressor_xz(void *in, size_t insz,
 	return SQFS_OK;
 }
 #define CAN_DECOMPRESS_XZ 1
+
+// The following implementation was adapted from  squashfs-tools
+// https://github.com/plougher/squashfs-tools.git
+
+#define LZMA_PROPS_SIZE 5
+#define LZMA_UNCOMP_SIZE 8
+#define LZMA_HEADER_SIZE (LZMA_PROPS_SIZE + LZMA_UNCOMP_SIZE)
+#define MEMLIMIT (32 * 1024 * 1024)
+
+static sqfs_err sqfs_decompressor_lzma(void *in, size_t insz,
+		void *out, size_t *outsz) {
+
+	lzma_stream strm = LZMA_STREAM_INIT;
+	int uncompressed_size = 0, res;
+	unsigned char lzma_header[LZMA_HEADER_SIZE];
+
+	res = lzma_alone_decoder(&strm, MEMLIMIT);
+	if(res != LZMA_OK) {
+		lzma_end(&strm);
+		return SQFS_ERR;
+	}
+
+	memcpy(lzma_header, in, LZMA_HEADER_SIZE);
+	uncompressed_size = lzma_header[LZMA_PROPS_SIZE] |
```

Comment:
Could just use `sqfs_swapin32`?

Reply:
What is the problem with current code and what is the benefit of using `sqfs_swapin32`

Reply:
We just already have our own endian-swapping functions, it doesn't seem like we should be adding more ad-hoc swapping code.